clients/mongo/config.go

```diff
@@ -0,0 +1,98 @@
+package mongo
+
+import (
+	"errors"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	common "github.com/tidepool-org/go-common"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// Config of the mongo database
+type Config struct {
+	Scheme                 string `json:"scheme"`
+	addresses              []string
+	TLS                    bool                          `json:"tls"`
+	Database               string                        `json:"database"`
+	Username               string                        `json:"-"`
+	Password               string                        `json:"-"`
+	Timeout                time.Duration                 `json:"timeout"`
+	OptParams              string                        `json:"optParams"`
+	WaitConnectionInterval time.Duration                 `json:"waitConnectionInterval"`
+	MaxConnectionAttempts  int64                         `json:"maxConnectionAttempts"`
+	Indexes                map[string][]mongo.IndexModel `json:"indexes"`
+}
+
+// FromEnv read the mongo config from the environment variables
+func (config *Config) FromEnv() {
+	config.Scheme, _ = os.LookupEnv("TIDEPOOL_STORE_SCHEME")
+	stringAddresses, _ := os.LookupEnv("TIDEPOOL_STORE_ADDRESSES")
+	addresses := []string{}
+	for _, str := range strings.Split(stringAddresses, ",") {
+		if str = strings.TrimSpace(str); str != "" {
+			addresses = append(addresses, str)
+		}
+	}
+	config.addresses = addresses
+	config.Username, _ = os.LookupEnv("TIDEPOOL_STORE_USERNAME")
+	config.Password, _ = os.LookupEnv("TIDEPOOL_STORE_PASSWORD")
+	config.Database, _ = os.LookupEnv("TIDEPOOL_STORE_DATABASE")
+	config.OptParams, _ = os.LookupEnv("TIDEPOOL_STORE_OPT_PARAMS")
+	tls, found := os.LookupEnv("TIDEPOOL_STORE_TLS")
+	config.TLS = found && tls == "true"
+
+	defaultTimeout := common.GetEnvironmentInt64("TIDEPOOL_STORE_DEFAULT_TIMEOUT", 2)
+	config.Timeout = time.Duration(defaultTimeout) * time.Second
+	waitConnectionInterval := common.GetEnvironmentInt64("TIDEPOOL_STORE_WAIT_CONNECTION_INTERVAL", 5)
+	config.WaitConnectionInterval = time.Duration(waitConnectionInterval) * time.Second
+	config.MaxConnectionAttempts = common.GetEnvironmentInt64("TIDEPOOL_STORE_MAX_CONNECTION_ATTEMPTS", 0)

```

Comment:
wondering if we should keep the default to 0
I would probably change it to 3 so that it gives us at least 15 seconds.
If 0 is the default value to make sure we do not stop the service even if DB is not available, that 's fine to keep it. But in that case I would propose to document it.
```suggestion
	config.MaxConnectionAttempts = common.GetEnvironmentInt64("TIDEPOOL_STORE_MAX_CONNECTION_ATTEMPTS", 0)
	// O is the default value to keep service running when db is not available
```

Reply:
I just aligned default values with what we defined in platform. Node/Typescript services don't have this max connection environment and will loop infinitely (like a zero value). 
Ok to change it but we will have to also change it in platform and maybe implement it for Node/Typescript services .